src/components/ont/runs/ONTFlowcell.vue

```diff
@@ -131,80 +133,81 @@ export default {
     },
     flowcellId: {
       get() {
-        const flowcell = this.currentRun().flowcell_attributes.find(
-          (flowcell) => flowcell.position == this.position,
-        )
-        if (flowcell) {
-          return flowcell.flowcell_id
+        if (this.currentRun) {
+          const flowcell = this.currentRun.flowcell_attributes.find(
+            (flowcell) => flowcell.position == this.position,
+          )
+          if (flowcell) {
+            return flowcell.flowcell_id
+          }
         }
         return ''
       },
       set(value) {
         this.setFlowcellId({ $event: value, position: this.position })
       },
     },
-    ...mapState({
-      poolTubeBarcode(state) {
-        const flowcell = state.currentRun.flowcell_attributes.find(
-          (flowcell) => flowcell.position == this.position,
-        )
-        if (flowcell) {
-          return flowcell.tube_barcode
-        }
-      },
-      /**Displays green if valid, red if invalid and no border if empty */
-      flowcell_id_field_colour() {
-        return this.isFlowcellIdExists
-          ? this.flowcellIdValidationError.length === 0
-            ? 'border-3 border-solid border-success'
-            : 'border-3 border-solid border-failure focus:border-failure'
-          : ''
-      },
-      /**Displays green if valid, red if invalid and no border if empty */
-      flowcell_barcode_field_colour() {
-        return this.isBarcodeExists
-          ? this.barcodeState
-            ? 'border-3 border-solid border-success'
-            : 'border-3 border-solid border-failure focus:border-failure'
-          : ''
-      },
-      /**
-       * green - if both flowcellId and barcode fields contain valid values
-       * red - if any of flowcellId and barcode fields contain invalid values
-       * white - if both flowcellId and barcode fields are empty
-       * yellow - if one of flowcellId and barcode fields are valid and other is empty
-       */
-      flowcell_bg_colour() {
-        if (!this.isFlowcellIdExists && !this.isBarcodeExists)
-          return 'border border-3 border-gray-300'
+    poolTubeBarcode() {
+      //This is keep poolTubeBarcode in sync with the Pinia store state  (option api way)
+      const ontRunStore = useOntRunsStore()
+      const flowcell = ontRunStore.currentRun.flowcell_attributes.find(
```

Comment:
same here. If you set currentRun in the store to be something like:

`{  flowcell_attributes: [ { position: 1, tube_barcode: '' }] }`

or you could return a blan flowcell. It is essentially adding a type so you don't need to do lot's of checking.